### PR TITLE
Added ability to specify the Mailgun API endpoint URL

### PR DESCRIPTION
**Pull request by mingfudesign**
Submitting changes allowing for configuration of the Mailgun API URL. Sometimes this ability is needed when a domain is provisioned in a non-US region, but the current Mailgun driver does not allow for this.

### CODE REVIEW

File: classes/email/driver/mailgun.php [hunk 1]
```diff
@@ -20,14 +20,24 @@ protected function _send()
 
 		$message = $this->build_message();
 
-		$mg = \Mailgun\Mailgun::create($this->config['mailgun']['key']);
-		
+		$config_mailgun = $this->config['mailgun'];
+
+		$endpoint = isset($config_mailgun['endpoint']) ? $config_mailgun['endpoint'] : null;
+
+		if (empty($endpoint)) {
```
Comment (WanWizard) [inline classes/email/driver/mailgun.php]: Coding standard: bracket on the next line please

**Review by WanWizard — CHANGES_REQUESTED** [review]
Can you also update the docs (/packages/email/usage.html) with the same remarks as in the README?

File: classes/email/driver/mailgun.php (hunk 1 above)
Comment (mingfudesign) [reply classes/email/driver/mailgun.php]: Sure thing. The source seems to be a mix of different standards, so we weren't sure what to do, and I ended up with some mix of the two. Will update and re-submit.

Comment (mingfudesign) [reply classes/email/driver/mailgun.php]: Would you like copyright dates updated?

From

``2010 - 2019 Fuel Development Team``

to 

``2010 - 2020 Fuel Development Team``

Comment (WanWizard) [reply classes/email/driver/mailgun.php]: > The source seems to be a mix of different standards

That's not good, I'll go through it.

> Would you like copyright dates updated?

No, it's ok, I've got a script for it, I'll change it to 2021 next month...

Comment (mingfudesign) [reply classes/email/driver/mailgun.php]: > > The source seems to be a mix of different standards
> 
> That's not good, I'll go through it.
> 
Well, I don't mean to overstate the mix.

In addition to making the requested change, I've changed the foreach brackets in the mailgun driver source to conform, and have updated the docs as requested. Will push changes later when able.

**Pull request merged**